azure/services/virtualnetworks/virtualnetworks.go

```diff
@@ -40,6 +40,7 @@ type VNetScope interface {
 	VNetSpec() azure.ResourceSpecGetter
 	ClusterName() string
 	IsVnetManaged() bool
+	UpdateSubnetCIDRs(string, []string)
```

Comment:
Just a note, I read through the existing implementation of `UpdateSubnetCIDRs` and it will silently fail (ignore) when you pass in a name string that doesn't actually match an existing subnet. Not sure if we want to add some reinforcement around there.

Reply:
yeah I think that's actually a good thing in this use case. Because the custom vnet might be shared for other purposes outside this cluster, it will likely have subnets we don't care about/don't recognize. We want to skip those and not fail when that happens.

Reply:
updated the comment to make it explicit that it's the desired behavior @jackfrancis